var/spack/repos/builtin/packages/nvshmem/package.py

```diff
@@ -73,12 +86,44 @@ def setup_build_environment(self, env):
 
             if self.spec.satisfies("^spectrum-mpi") or self.spec.satisfies("^openmpi"):
                 env.set("NVSHMEM_MPI_IS_OMPI", "1")
+                if self.spec.satisfies("^openmpi") and self.spec.satisfies("^openmpi +pmix"):
+                    env.set("NVSHMEM_PMIX_SUPPORT", "1")
+                else:
+                    env.set("NVSHMEM_PMIX_SUPPORT", "0")
             else:
                 env.set("NVSHMEM_MPI_IS_OMPI", "0")
+                env.set("NVSHMEM_PMIX_SUPPORT", "0")
 
         if "+shmem" in self.spec:
             env.set("NVSHMEM_SHMEM_SUPPORT", "1")
             env.set("SHMEM_HOME", self.spec["mpi"].prefix)
+            env.set("NVSHMEM_LSHMEM", "-loshmem")
+
+        if "+gpu_initiated_support" in self.spec:
+            if self.spec.satisfies("@2.9:"):
+#                env.set("NVSHMEM_IBGDA_SUPPORT", "1")
+                env.set("NVSHMEM_IBGDA_SUPPORT_GPUMEM_ONLY", "1")
+#-L$(NON_STANDARD_MLX5_LOCATION) -lmlx5
+            else:
+                env.set("NVSHMEM_GPUINITIATED_SUPPORT", "1")
+
+        if "+ibrc" in self.spec:
+            env.set("NVSHMEM_IBRC_SUPPORT", "1")
 
+    @when("@2.6:")
```

Comment:
Is the @2.9 inside of setup_run_environment extraneous? I thought the @when("@2.6:") would only run this function for 2.6? (Just getting a grasp of spack syntax).